### PR TITLE
GUM-40: Add quiz 

### PR DESCRIPTION
**Pull request by katyakostikova**

### CODE REVIEW

File: src/screens/quiz/components/quiz-item/quiz-item.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    answerId: string,
+    prevAnswerId: string
+  ) => void;
+}
+
+const QuizItem: React.FC<IQuizItemProps> = ({
+  item,
+  onChangeQuestionAnswer,
+}) => {
+  const [value, setValue] = useState('0');
```
Comment (AleksandrTermenzhy) [inline src/screens/quiz/components/quiz-item/quiz-item.tsx:20]: Do we need `'0'` here or empty string can fit too?

File: src/screens/quiz/components/quiz-item/quiz-item.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  return (
+    <View>
+      <Heading level={HeadingLevel.H5} style={styles.question}>
+        {item.question}
+      </Heading>
+      <View style={styles.answersContainer}>
+        <RadioButton.Group onValueChange={handleChangeAnswer} value={value}>
+          {item.answers.map((answer, index) => (
+            <View key={answer.id} style={styles.answerContainer}>
+              <RadioButton value={index.toString()} />
```
Comment (AleksandrTermenzhy) [inline src/screens/quiz/components/quiz-item/quiz-item.tsx]:
```suggestion
              <RadioButton value={index?.toString()} />
```

File: src/screens/quiz/quiz.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+const QuizScreen: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const navigation = useAppNavigation();
+  const { quizQuestions } = useAppSelector(state => state.quiz);
+  const pagerRef = useRef<PagerView | null>(null);
+  const [currentStep, setCurrentStep] = useState(0);
+
+  const isLastStep = currentStep === quizQuestions.length - 1;
+
+  const saveAnswer = (
```
Comment (AleksandrTermenzhy) [inline src/screens/quiz/quiz.tsx]:
```suggestion
  const handleAnswerChange = (
```

File: src/screens/quiz/components/quiz-item/quiz-item.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import { RadioButton } from 'react-native-paper';
+
+import { HeadingLevel } from 'src/common/enums';
+import { IQuizQuestion } from 'src/common/types';
+import { Heading, Text } from 'src/components';
+import styles from './styles';
+
+interface IQuizItemProps {
+  item: IQuizQuestion;
+  onChangeQuestionAnswer: (
```
Comment (AleksandrTermenzhy) [inline src/screens/quiz/components/quiz-item/quiz-item.tsx]:
```suggestion
  onAnswerChange: (
```

**Review by AleksandrTermenzhy — APPROVED** [review]

File: src/screens/profile/components/quiz-info/styles.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  quizHeader: {
+    marginVertical: 10,
+  },
+  quizContent: {
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+});
+
+export default styles;
```
Comment (R-Mielamud) [inline src/screens/profile/components/quiz-info/styles.ts]: Please wait until #48 is merged and replace with `useStyles` hook.

File: src/screens/profile/profile.tsx [hunk 6]
```diff
@@ -64,6 +64,7 @@ const ProfileScreen: React.FC = () => {
   };
 
   const { careerExperience } = useAppSelector(state => state.experience);
+  const { user } = useAppSelector(state => state.auth);
```
Comment (R-Mielamud) [inline src/screens/profile/profile.tsx]: Use one selector and form an object inside it.

File: src/screens/quiz/styles.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    top: 0,
+    alignSelf: 'center',
+  },
+  completeButton: {
+    borderWidth: 2,
+    borderColor: AppColor.ACCENT,
+  },
+});
+
+export default styles;
```
Comment (R-Mielamud) [inline src/screens/quiz/styles.ts]: Also replace with `useStyles`/`useColor` when #48 is merged

File: src/store/quiz/slice.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      state.quizResults = payload;
+    });
+    builder.addCase(saveQuizAnswers, (state, { payload }) => {
+      const { questionId, answerId, prevAnswerId } = payload;
+      const questionIndex = state.quizQuestions.findIndex(
+        item => item.id === questionId
+      );
+      state.quizQuestions[questionIndex].answers[+prevAnswerId].isSelected =
+        false;
+      state.quizQuestions[questionIndex].answers[+answerId].isSelected = true;
```
Comment (R-Mielamud) [inline src/store/quiz/slice.ts]: Can you simplify this? As I know, all ids are UUIDs

**Review by R-Mielamud — APPROVED** [review]

**Pull request merged**